tests/Committee/CommitteeManagementAuthorityTest.php

```diff
@@ -14,53 +15,121 @@
             '/comites/%s/%s',
             LoadAdherentData::COMMITTEE_1_UUID,
             'comite-lille-beach'
         ));
-        $urlGenerator->expects($this->once())->method('generate')->willReturn(sprintf(
+
+        $committeeManagementAuthority = new CommitteeManagementAuthority($manager, $urlGenerator, $mailer);
+        $committeeManagementAuthority->approve($committee);
+    }
+
+    public function testNotifyReferentsForApproval()
+    {
+        $committee = $this->createCommittee(LoadAdherentData::COMMITTEE_1_UUID, 'Paris 8e');
+        $animator = $this->createAnimator(LoadAdherentData::ADHERENT_3_UUID);
+        $referent = $this->createMock(Adherent::class);
+
+        $referents = new AdherentCollection([$referent]);
+        $manager = $this->createManager($committee, $animator, $referents);
+
+        $mailer = $this->createMock(MailjetService::class);
+        $mailer->expects($this->at(0))
+            ->method('sendMessage')
+            ->with($this->isInstanceOf(CommitteeApprovalReferentMessage::class));
+
+        $urlGenerator = $this->createMock(CommitteeUrlGenerator::class);
+        $urlGenerator->expects($this->any())->method('generate')->willReturn(sprintf(
             '/espace-adherent/contacter/%s?from=%s&id=%s',
             LoadAdherentData::ADHERENT_3_UUID,
             'committee',
             LoadAdherentData::COMMITTEE_1_UUID
         ));
 
         $committeeManagementAuthority = new CommitteeManagementAuthority($manager, $urlGenerator, $mailer);
-        $committeeManagementAuthority->approve($committee);
+        $committeeManagementAuthority->notifyReferentsForApproval($committee);
+    }
+
+    /**
+     * @expectedException \AppBundle\Committee\MultipleReferentsFoundException
+     */
+    public function testNotifyReferentsForApprovalWithMultipleReferents()
+    {
+        $committee = $this->createCommittee(LoadAdherentData::COMMITTEE_1_UUID, 'Paris 8e');
+        $animator = $this->createAnimator(LoadAdherentData::ADHERENT_3_UUID);
+        $referent = $this->createMock(Adherent::class);
+
+        $referents = new AdherentCollection([$referent, $animator]);
+
+        $manager = $this->createManager($committee, $animator, $referents);
+
+        $mailer = $this->createMock(MailjetService::class);
+        // ensure no mail is sent
+        $mailer->expects($this->never())->method('sendMessage')->with($this->anything());
+
+        $urlGenerator = $this->createMock(CommitteeUrlGenerator::class);
+        $urlGenerator->expects($this->never())->method('generate')->with($this->anything());
+
+        $committeeManagementAuthority = new CommitteeManagementAuthority($manager, $urlGenerator, $mailer);
+        $committeeManagementAuthority->notifyReferentsForApproval($committee);
+    }
+
+    private function createCommittee(string $uuid, string $cityName): Committee
+    {
+        $committeeUuid = $this->createMock(Uuid::class);
```

Comment:
You should use the real ` Uuid`  class here. This is a value object so it's lightweight compared to a mock object.

Reply:
:+1: